reactor-core/src/test/java/reactor/core/publisher/FluxHandleTest.java

```diff
@@ -685,6 +687,18 @@ public void nextAfterErrorFused() {
 		                                                    .hasMessage("Cannot emit after a complete or error"));
 	}
 
+    @Test
+    public void runtimeExceptionFused() {
+        // Check issue that for fuseable and Exception thrown inside handle it did not propagate signal -> hanging flux
+        //  and throws IllegalStateException: Timeout on blocking read
+        Assertions.assertThatThrownBy(() -> {
+            Flux.range(0, 10)
+                .handle((v, sink) -> {
+                    throw new NullPointerException("boom");
+                })
+                .blockLast(Duration.ofSeconds(1));
+        }).isInstanceOf(NullPointerException.class).hasMessageContaining("boom");
```

Comment:
assuming you switch to `assertThatNullPointerException...` above, this line can be simplified to `}).withMessage("boom");`
```